### PR TITLE
KAA-1470: [Java SDK] Failover strategy doesn't work correctly

### PR DESCRIPTION
**Pull request by Kirill380**
…ss to Internet

### CODE REVIEW

**Review by garniy — APPROVED** [review]

File: client/client-multi/client-java-desktop/src/main/java/org/kaaproject/kaa/client/connectivity/PingConnectivityChecker.java [hunk 1]
```diff
@@ -22,33 +22,41 @@
...
-  public PingConnectivityChecker(String host) {
-    this.host = host;
+  public PingConnectivityChecker(String urlAddress) {
+    this.urlAddress = urlAddress;
   }
 
   @Override
   public boolean checkConnectivity() {
     try {
-      return InetAddress.getByName(host).isReachable(CONNECTION_TIMEOUT_MS);
```
Comment (agulenko) [inline client/client-multi/client-java-desktop/src/main/java/org/kaaproject/kaa/client/connectivity/PingConnectivityChecker.java:48]: Not critical, but why not just:
 try {
        try (Socket soc = new Socket()) {
            soc.connect(new InetSocketAddress(addr, openPort), timeOutMillis);
        }
        return true;
    } catch (IOException ex) {
        return false;
    }

**Review by vtkhir — APPROVED** [review]

**Review by garniy — APPROVED** [review]

File: server/common/log-shared/src/main/java/org/kaaproject/kaa/server/common/log/shared/avro/gen/RecordHeader.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
  *
  * DO NOT EDIT DIRECTLY
  */
+
```
Comment (akokhanovskyi) [inline server/common/log-shared/src/main/java/org/kaaproject/kaa/server/common/log/shared/avro/gen/RecordHeader.java:22]: This file should not have changed, I suppose

File: client/client-multi/client-java-desktop/src/test/java/org/kaaproject/kaa/client/connectivity/PingConnectivityCheckerTest.java [hunk 3]
```diff
@@ -23,13 +23,13 @@
...
   public void checkConnectivityToWrongHostTest() {
-    PingConnectivityChecker pingConnectivityChecker = new PingConnectivityChecker("some-wrong-host=with-dashes-in-name");
+    PingConnectivityChecker pingConnectivityChecker = new PingConnectivityChecker("some-wrong-url=");
     Assert.assertFalse(pingConnectivityChecker.checkConnectivity());
   }
 
   @Test
   public void checkConnectivityForLocalHostTest() {
-    PingConnectivityChecker pingConnectivityChecker = new PingConnectivityChecker("localhost");
+    PingConnectivityChecker pingConnectivityChecker = new PingConnectivityChecker("http://google.com");
```
Comment (akokhanovskyi) [inline client/client-multi/client-java-desktop/src/test/java/org/kaaproject/kaa/client/connectivity/PingConnectivityCheckerTest.java]: This does not match the test name. Please update the test name.

**Review by akokhanovskyi — CHANGES_REQUESTED** [review]

File: server/common/log-shared/src/main/java/org/kaaproject/kaa/server/common/log/shared/avro/gen/RecordHeader.java (hunk 2 above)
Comment (Kirill380) [reply server/common/log-shared/src/main/java/org/kaaproject/kaa/server/common/log/shared/avro/gen/RecordHeader.java:22]: I  separated license from class, this does not affect code, just for pretty look.

**Review by agulenko — APPROVED** [review]

**Review by garniy — APPROVED** [review]

**Review by akokhanovskyi — APPROVED** [review]

**Pull request merged**